src/utils.js

```diff
@@ -52,8 +52,16 @@ module.exports.createNetworkId = function() {
 
 module.exports.delimiter = '---';
 
+function sanitizeKey(key) {
+  return key.replace(/\./g, '!').replace(/\#/g, '@').replace(/\[/g, '{').replace(/\]/g, '}'); 
+}
+
+function desanitizeKey(key) {
+  return key.replace(/\!/g, '.').replace(/\@/g, '#').replace(/\{/g, '[').replace(/\}/g, ']'); 
+}
+
 module.exports.childSchemaToKey = function(schema) {
-  var key = (schema.selector || '')
+  var key = sanitizeKey(schema.selector || '')
```

Comment:
@machenmusik this shouldn't need to here due to this line: https://github.com/haydenjameslee/networked-aframe/blob/master/src/adapters/FirebaseWebRtcAdapter.js#L175

Are there cases that this doesn't solve? It would be best if the firebase restrictions didn't leak out into all keys.